Assets/Editor/ExportOBJ.cs

```diff
@@ -0,0 +1,158 @@
+﻿using UnityEngine;
+using UnityEditor;
+using System.Collections;
+using System.IO;
+using System.Text;
+
+public class ObjExporterScript
+{
+	private static int StartIndex = 0;
+
+	public static void Start()
+	{
+		StartIndex = 0;
+	}
+	public static void End()
+	{
+		StartIndex = 0;
+	}
+
+	public static string MeshToString( MeshFilter mf, Transform t )
+	{
+		Vector3 s = t.localScale;
+		Vector3 p = t.localPosition;
+		Quaternion r = t.localRotation;
+
+		int numVertices = 0;
+		Mesh m = mf.sharedMesh;
+		if ( !m )
+		{
+			return "####Error####";
+		}
+		Material[] mats = mf.renderer.sharedMaterials;
+
+		StringBuilder sb = new StringBuilder();
+
+		foreach ( Vector3 vv in m.vertices )
+		{
+			Vector3 v = t.TransformPoint( vv );
+			numVertices++;
+			sb.Append( string.Format( "v {0} {1} {2}\n", v.x, v.y, -v.z ) );
```

Comment:
[This article](http://www.dotnetperls.com/stringbuilder) talks about the performance of string builders down towards the bottom. For the most part, if you're going to do a lot of appending, then `StringBuilder.Append()` is going to be faster than +=. We can still speed this up, though, by telling the string builder from the beginning how big the final string is going to be. Since we know how many vertices and faces there are, we should be able to do that calculation before actually building the string.